library/math/numbersLine/src/components/MoveableElement.tsx

```diff
@@ -1,30 +1,47 @@
-import Moveable from "react-moveable";
-
+import { useNumbersLineContext } from "../context/numbersLineContext";
+import Moveable, { OnResize, OnResizeEnd } from "react-moveable";
+import { IElement } from "../type/elements";
 interface IProps {
   targetRef: any;
+  element: IElement;
+  unit: number;
 }
 
-const MoveableElement = ({ targetRef }: IProps) => {
+const MoveableElement = ({ targetRef, element, unit }: IProps) => {
+  const { windowWidth, dragElements, setDragElements } = useNumbersLineContext();
+
+  const hideValueElement = () => {
+    let newElements = dragElements.map((item: IElement) => (item.id === element.id ? { ...item, hideNumber: true } : item));
+    setDragElements(newElements);
+  };
+  const changeElementValue = (e: OnResizeEnd) => {
+    let newValue = Math.round(e.lastEvent.width / unit);
+    let newElements = dragElements.map((item: IElement) => (item.id === element.id ? { ...item, value: newValue } : item));
+    setDragElements(newElements);
+    e.target.style.width = `${newValue * unit}px`;
+  };
+
+  const updateTransform = (e: OnResize) => {
+    if (!(parseFloat(e.target.style.width) / unit < 1 && e.dist[0] < 0) && !(parseFloat(e.target.style.width) > windowWidth - 4 * 16 && e.dist[0] > 0)) {
```

Comment:
creat const of padding. use it for padding and for now. not calc with number. because, if you want to change the padding you shall have to change also this row. instead change just a const

Reply:
ho i do something like this and i use this in other page
i show you in the next review what i do...